cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/store/AppMetadataStoreTest.java

```diff
@@ -210,113 +207,71 @@ public void apply() throws Exception {
         metadataStoreDataset.recordProgramStop(program, runId.getId(), RunIds.getTime(runId, TimeUnit.SECONDS),
                                                ProgramRunStatus.KILLED,
                                                null, AppFabricTestHelper.createSourceId(killedSourceId));
-        resultRecord.set(metadataStoreDataset.getRun(program, runId.getId()));
+        Assert.assertEquals(expectedRunStatus, metadataStoreDataset.getRun(program, runId.getId()).getStatus());
       }
     });
-    Assert.assertEquals(expectedRunStatus, resultRecord.get().getStatus());
   }
 
   private void assertPersistedStatus(final AppMetadataStore metadataStoreDataset, TransactionExecutor txnl,
-                                     @Nullable ProgramRunStatus expectedRunStatus,
-                                     List<ProgramRunStatus> recordStatuses)
-    throws Exception {
+                                     @Nullable final ProgramRunStatus expectedRunStatus,
+                                     final List<ProgramRunStatus> recordStatuses) throws Exception {
     // Add some run records
     ApplicationId application = NamespaceId.DEFAULT.app("app");
     final ProgramId program = application.program(ProgramType.WORKFLOW, "program");
     final RunId runId = RunIds.generate(10000);
     final AtomicLong sourceId = new AtomicLong();
-    final Map<ProgramRunStatus, Runnable> statusRecordMap =
-      getStatusRecordMap(metadataStoreDataset, program, runId, sourceId);
-    // Get the record status runnable for each program run status
-    final List<Runnable> recordRunnables = Lists.transform(recordStatuses, new Function<ProgramRunStatus, Runnable>() {
-      @Nullable
-      @Override
-      public Runnable apply(ProgramRunStatus runStatus) {
-        return statusRecordMap.get(runStatus);
-      }
-    });
     // Execute all record status runnable and assert the result
-    final AtomicReference<RunRecordMeta> resultRecord = new AtomicReference<>();
     txnl.execute(new TransactionExecutor.Subroutine() {
       @Override
       public void apply() throws Exception {
-        for (Runnable runnable : recordRunnables) {
-          if (runnable != null) {
-            runnable.run();
-          }
+        for (ProgramRunStatus runStatus : recordStatuses) {
```

Comment:
this isn't what I meant. These two methods are just a more complicated way of executing a transaction and running a couple recordProgramXYZ calls. Remove both of these methods and just call metadataStoreDataset.recordProgramXYZ. There is no value in adding two methods that just turn a ProgramRunStatus into a recordProgram call, it just makes it harder to understand what is happening.